src/Aspire.Hosting.Azure.Storage/AzureStorageExtensions.cs

```diff
@@ -280,22 +313,40 @@ public static IResourceBuilder<AzureStorageEmulatorResource> WithApiVersionCheck
             return Task.CompletedTask;
         });
     }
-
     /// <summary>
     /// Creates a builder for the <see cref="AzureBlobStorageResource"/> which can be referenced to get the Azure Storage blob endpoint for the storage account.
     /// </summary>
-    /// <param name="builder">The <see cref="IResourceBuilder{T}"/> for <see cref="AzureStorageResource"/>/</param>
+    /// <param name="builder">The <see cref="IResourceBuilder{T}"/> for <see cref="AzureStorageResource"/>.</param>
     /// <param name="name">The name of the resource.</param>
     /// <returns>An <see cref="IResourceBuilder{T}"/> for the <see cref="AzureBlobStorageResource"/>.</returns>
+    [Obsolete("Use AddBlobService on IResourceBuilder<AzureStorageResource> instead.")]
     public static IResourceBuilder<AzureBlobStorageResource> AddBlobs(this IResourceBuilder<AzureStorageResource> builder, [ResourceName] string name)
     {
         ArgumentNullException.ThrowIfNull(builder);
         ArgumentException.ThrowIfNullOrEmpty(name);
 
+        return AddBlobService(builder, name);
+    }
+
+    /// <summary>
+    /// Creates a builder for the <see cref="AzureBlobStorageResource"/> which can be referenced to get the Azure Storage blob endpoint for the storage account.
+    /// </summary>
+    /// <param name="builder">The <see cref="IResourceBuilder{T}"/> for <see cref="AzureStorageResource"/>.</param>
+    /// <param name="name">The name of the resource.</param>
+    /// <returns>An <see cref="IResourceBuilder{T}"/> for the <see cref="AzureBlobStorageResource"/>.</returns>
+    public static IResourceBuilder<AzureBlobStorageResource> AddBlobService(this IResourceBuilder<AzureStorageResource> builder, [ResourceName] string? name = null)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        
+        name ??= builder.Resource.Name + "-blobs";
+
         var resource = new AzureBlobStorageResource(name, builder.Resource);
+        builder.Resource.BlobStorageResource = resource;
```

Comment:
Do we want to overwrite this every time this method is called?

Reply:
Test added. REsult is correct.